src/components/ContactsAdd.js

```diff
@@ -1,39 +1,156 @@
-import { useState } from "react"
-import { useNavigate } from "react-router-dom";
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 
-function ContactsAdd(props) {
+function ContactsAdd({ contacts, setContacts }) {
+	const [addContactFormEmpty, setAddContactFormEmpty] = useState({
+		firstName: '',
+		lastName: '',
+		street: '',
+		city: '',
+		email: '',
+		linkedIn: '',
+		twitter: '',
+	});
 
-  // setContacts and contacts must be passed as props
-  // to this component so new contacts can be added to the
-  // state
-  const { setContacts, contacts } = props
+	const [addContactForm, setAddContactForm] = useState(addContactFormEmpty);
 
-  //TODO: Implement controlled form
-  //send POST to json server on form submit
+	let navigate = useNavigate();
 
-  return (
-    <form className="form-stack contact-form">
-      <h2>Create Contact</h2>
+	const onFirstNameChange = (event) => {
+		setAddContactForm({ ...addContactForm, firstName: event.target.value });
+	};
 
-      <label htmlFor="firstName">First Name</label>
-      <input id="firstName" name="firstName" type="text" required />
+	const onLastNameChange = (event) => {
+		setAddContactForm({ ...addContactForm, lastName: event.target.value });
+	};
 
-      <label htmlFor="lastName">Last Name:</label>
-      <input id="lastName" name="lastName" type="text" required/>
+	const onStreetChange = (event) => {
+		setAddContactForm({ ...addContactForm, street: event.target.value });
+	};
 
-      <label htmlFor="street">Street:</label>
-      <input id="street" name="street" type="text" required/>
+	const onCityChange = (event) => {
+		setAddContactForm({ ...addContactForm, city: event.target.value });
+	};
 
-      <label htmlFor="city">City:</label>
-      <input id="city" name="city" type="text" required/>
+	const onEmailChange = (event) => {
+		setAddContactForm({ ...addContactForm, email: event.target.value });
+	};
 
-      <div className="actions-section">
-        <button className="button blue" type="submit">
-          Create
-        </button>
-      </div>
-    </form>
-  )
+	const onLinkedInChange = (event) => {
+		setAddContactForm({ ...addContactForm, linkedIn: event.target.value });
+	};
+
+	const onTwitterChange = (event) => {
```

Comment:
u could have used one function only to handle the form input, coz this is your chance to practice forms styles in react